### DISCUSSION TITLE
How to clear the consumer config in JUnit

### DISCUSSION THREAD

**Discussion by ahmadirfaan** [sofastack/sofa-rpc#1345] [Q&A]
Hi All, 

I having problem about running some test in the SOFABoot, when running test @SpringBootTest

The error is that the consumer config that I created is exceed the value of "consumerRepeatedReferenceLimit" or "com.alipay.sofa.rpc.consumer.repeated.reference.limit".

I have trying to set the consumerRepeatedReference in application.properties but limit of consumer config is not changing??

![image](https://github.com/sofastack/sofa-rpc/assets/49211264/a35b5be4-7700-4743-a404-d65d5c1f1375)

**Comment by khotyn** [comment]
@ahmadirfaan What version of SOFABoot and SOFARPC do you have?

**Reply by ahmadirfaan** [reply]
my sofa-rpc version is

sofa-rpc-all: 5.5.7 

and version of my sofa boot project is sofa-boot-enterprise, I think not using this public repo

**Comment by OrezzerO** [comment]
There are several approaches to solve it. 
1. You can make consumer in one Spring bean, and use then by autowire the bean. So consumers will not be referred more than once.
2. You can set consumer Repeated Reference to -1 ,to allow refer consumer more than once.

By the way, how do you config consumerRepeatedReference ?  Can you provider more details?

**Reply by ahmadirfaan** [reply]
from the default project, I think my project not config the consumerRepeatedReference, but after debugging, I think the value is always 3. I have tried set the consumerRepeatedReferencen from the application.properties

from the answer number 2: 
- the consumer repeated reference is config by rpc client right??

**Reply by OrezzerO** [reply]
You can set consumerRepeatedReference to -1 in your test code.
